src/Functions/vectorFunctions.cpp

```diff
@@ -0,0 +1,1067 @@
+                    continue;
+                }
+
+                ColumnWithTypeAndName left_type{res_type, {}};
+                ColumnWithTypeAndName right_type{elem_pow->getResultType(), {}};
+                auto plus_elem = plus->build({left_type, right_type});
+                res_type = plus_elem->getResultType();
+            }
+            catch (DB::Exception & e)
+            {
+                e.addMessage("While executing function {} for tuple element {}", getName(), i);
+                throw;
+            }
+        }
+
+        auto divide = FunctionFactory::instance().get("divide", context);
+        ColumnWithTypeAndName one{std::make_shared<DataTypeUInt8>(), {}};
+        auto div_elem = divide->build({one, p_column});
+        ColumnWithTypeAndName inv_p_column{div_elem->getResultType(), {}};
+        return pow->build({ColumnWithTypeAndName{res_type, {}}, inv_p_column})->getResultType();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        const auto * cur_tuple = checkAndGetDataType<DataTypeTuple>(arguments[0].type.get());
+        const auto & cur_types = cur_tuple->getElements();
+        auto cur_elements = getTupleElements(*arguments[0].column);
+
+        size_t tuple_size = cur_elements.size();
+        if (tuple_size == 0)
+            return DataTypeUInt8().createColumnConstWithDefaultValue(input_rows_count);
+
+        const auto & p_column = arguments[1];
+        auto abs = FunctionFactory::instance().get("abs", context);
+        auto pow = FunctionFactory::instance().get("pow", context);
+        auto plus = FunctionFactory::instance().get("plus", context);
+        ColumnWithTypeAndName res;
+        for (size_t i = 0; i < tuple_size; ++i)
+        {
+            ColumnWithTypeAndName cur{cur_elements[i], cur_types[i], {}};
+            auto elem_abs = abs->build(ColumnsWithTypeAndName{cur});
+            cur.column = elem_abs->execute({cur}, elem_abs->getResultType(), input_rows_count);
+            cur.type = elem_abs->getResultType();
+
+            auto elem_pow = pow->build(ColumnsWithTypeAndName{cur, p_column});
+
+            ColumnWithTypeAndName column;
+            column.type = elem_pow->getResultType();
+            column.column = elem_pow->execute({cur, p_column}, column.type, input_rows_count);
+
+            if (i == 0)
+            {
+                res = std::move(column);
+            }
+            else
+            {
+                auto plus_elem = plus->build({res, column});
+                auto res_type = plus_elem->getResultType();
+                res.column = plus_elem->execute({res, column}, res_type, input_rows_count);
+                res.type = res_type;
+            }
+        }
+
+        auto divide = FunctionFactory::instance().get("divide", context);
+        ColumnWithTypeAndName one{DataTypeUInt8().createColumnConst(input_rows_count, 1)->convertToFullColumnIfConst(),
```

Comment:
Do we really need `convertToFullColumnIfConst` ?

Reply:
Maybe not. I will delete the conversion.